Sources/OpenAPIRuntime/Interface/ErrorHandlingMiddleware.swift

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+
+/// An opt-in error Handling middleware that converts an error to a HTTP response.
+///
+/// Inclusion of  ``ErrorHandlingMiddleware`` should be accompanied by confirming errors to ``HTTPResponseConvertible``  protocol.
+/// Undocumented errors, i.e, errors not confriming to ``HTTPResponseConvertible`` are converted to a response with 500 status code.
+///
+/// Example usage
+/// 1. Create an error type that conforms to HTTPResponseConvertible protocol:
+///
+/// ```swift
+/// extension MyAppError: HTTPResponseConvertible {
+///    var httpStatus: HTTPResponse.Status {
+///    switch self {
+///        case .invalidInputFormat:
+///            .badRequest
+///        case .authorizationError:
+///            .forbidden
+///        }
+///    }
+/// }
+/// ```
+///
+/// 2. Opt in to the ``ErrorHandlingMiddleware``  while registering the handler:
+///
+/// ```swift
+/// let handler = try await RequestHandler()
+/// try handler.registerHandlers(on: transport, middlewares: [ErrorHandlingMiddleware()])
+/// ```
+/// - Note: The placement of ``ErrorHandlingMiddleware`` in the middleware chain is important.
+/// It should be determined based on the specific needs of each application.
+/// Consider the order of execution and dependencies between middlewares.
+public struct ErrorHandlingMiddleware: ServerMiddleware {
+    public func intercept(
```

Comment:
```suggestion
public struct ErrorHandlingMiddleware: ServerMiddleware {

    public init() {}

    public func intercept(
```

We need a public initializer here.

And please add doc comments to both.